// @jahia/ui-extender/dist/esm/actions/core/DisplayActions.js
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["target", "filter"];
import React from 'react';
import PropTypes from 'prop-types';
import { registry } from "../../registry";
import { DisplayAction } from "./DisplayAction";
export var DisplayActions = function DisplayActions(_ref) {
  var target = _ref.target,
    filter = _ref.filter,
    others = _objectWithoutProperties(_ref, _excluded);
  var actionsToDisplay = [];
  var targets = [];
  if (Array.isArray(target)) {
    targets = target;
  } else {
    targets.push(target);
  }
  targets.forEach(function (t) {
    var actions = registry.find({
      type: 'action',
      target: t
    }).filter(filter ? filter : function () {
      return true;
    }).map(function (action) {
      return /*#__PURE__*/React.createElement(DisplayAction, _extends({}, others, {
        key: action.key,
        target: t,
        actionKey: action.key
      }));
    });
    actionsToDisplay.push.apply(actionsToDisplay, _toConsumableArray(actions));
  });
  return /*#__PURE__*/React.createElement(React.Fragment, null, actionsToDisplay);
};
DisplayActions.defaultProps = {
  filter: null
};
DisplayActions.propTypes = {
  /**
       * The action context
       */
  context: PropTypes.object,
  /**
   * The target from which the items will be selected
   */
  target: PropTypes.string.isRequired,
  /**
   * The render component
   */
  render: PropTypes.func.isRequired,
  /**
   * The loading component
   */
  loading: PropTypes.func,
  /**
   * Additional filter function
   */
  filter: PropTypes.func
};
