// @jahia/ui-extender/dist/js/actions/core/DisplayActions.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DisplayActions = void 0;
var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _registry = require("../../registry");
var _DisplayAction = require("./DisplayAction");
var _excluded = ["target", "filter"];
var DisplayActions = function DisplayActions(_ref) {
  var target = _ref.target,
    filter = _ref.filter,
    others = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var actionsToDisplay = [];
  var targets = [];
  if (Array.isArray(target)) {
    targets = target;
  } else {
    targets.push(target);
  }
  targets.forEach(function (t) {
    var actions = _registry.registry.find({
      type: 'action',
      target: t
    }).filter(filter ? filter : function () {
      return true;
    }).map(function (action) {
      return /*#__PURE__*/_react["default"].createElement(_DisplayAction.DisplayAction, (0, _extends2["default"])({}, others, {
        key: action.key,
        target: t,
        actionKey: action.key
      }));
    });
    actionsToDisplay.push.apply(actionsToDisplay, (0, _toConsumableArray2["default"])(actions));
  });
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, actionsToDisplay);
};
exports.DisplayActions = DisplayActions;
DisplayActions.defaultProps = {
  filter: null
};
DisplayActions.propTypes = {
  /**
       * The action context
       */
  context: _propTypes["default"].object,
  /**
   * The target from which the items will be selected
   */
  target: _propTypes["default"].string.isRequired,
  /**
   * The render component
   */
  render: _propTypes["default"].func.isRequired,
  /**
   * The loading component
   */
  loading: _propTypes["default"].func,
  /**
   * Additional filter function
   */
  filter: _propTypes["default"].func
};
