// @jahia/moonstone/dist/CheckboxItem-DNfJwvmi.js
import { jsx as n, jsxs as b } from "react/jsx-runtime";
import I, { useRef as R, useState as _, createContext as T } from "react";
import { c as d } from "./clsx-OuTLNxxd.js";
import './CheckboxItem.css';const h = I.forwardRef(({
  children: o = "",
  component: e,
  variant: s = "body",
  weight: l = "default",
  className: a = "",
  hasLineThrough: r = !1,
  isItalic: i = !1,
  isUpperCase: t = !1,
  isNowrap: f = !1,
  ...m
}, c) => o ? /* @__PURE__ */ n(
  e || "p",
  {
    ref: c,
    className: d(
      "moonstone-typography",
      `moonstone-variant_${s}`,
      `moonstone-weight_${l}`,
      a,
      { "moonstone-nowrap": f },
      { "moonstone-italic": i },
      { "moonstone-upperCase": t },
      { "moonstone-lineThrough": r }
    ),
    ...m,
    children: o
  }
) : null);
h.displayName = "Typography";
const N = (o) => o.charAt(0).toUpperCase() + o.slice(1), u = ({ className: o, checked: e = !1, indeterminate: s = !1, size: l = "default", isDisabled: a, isReadOnly: r, onChange: i, value: t, ...f }) => {
  const m = R(null);
  return /* @__PURE__ */ b("div", { className: d("moonstone-checkbox", o), children: [
    /* @__PURE__ */ n(
      "input",
      {
        ...f,
        ref: m,
        className: d("moonstone-checkbox_input", `moonstone-checkbox_size${N(l)}`),
        type: "checkbox",
        value: t,
        checked: e,
        disabled: a,
        "aria-readonly": r,
        "aria-checked": s ? "mixed" : e,
        onChange: (c) => typeof i == "function" && i(c, t, m.current?.checked)
      }
    ),
    /* @__PURE__ */ n("svg", { className: d("moonstone-checkbox_icon", `moonstone-checkbox_size${N(l)}`), viewBox: "0 0 21 21", children: s ? /* @__PURE__ */ n("path", { d: "M4.5 10.5L16.5 10.5", strokeLinecap: "round" }) : /* @__PURE__ */ n("path", { d: "M5 10.75L8.5 14.25L16 6", strokeLinecap: "round" }) })
  ] });
};
u.displayName = "ControlledCheckbox";
const w = ({ defaultChecked: o = !1, onChange: e, value: s, ...l }) => {
  const [a, r] = _(o);
  return /* @__PURE__ */ n(
    u,
    {
      ...l,
      value: s,
      checked: a,
      onChange: (i) => {
        const t = !a;
        r(t), typeof e == "function" && e(i, s, t);
      }
    }
  );
};
w.displayName = "UncontrolledCheckbox";
const $ = ({ checked: o, ...e }) => typeof o > "u" ? /* @__PURE__ */ n(w, { ...e }) : /* @__PURE__ */ n(u, { checked: o, onChange: e.onChange, ...e });
$.displayName = "Checkbox";
const D = T(void 0), C = ({ className: o, id: e, value: s, label: l, description: a, isDisabled: r, isReadOnly: i, onChange: t, name: f, ...m }) => {
  const c = I.useContext(D), p = typeof c > "u" ? r : c.isDisabled, x = typeof c > "u" ? i : c.isReadOnly, L = typeof c > "u" ? f : c.name, U = R(null);
  return /* @__PURE__ */ b(
    h,
    {
      ref: U,
      className: d("moonstone-checkboxItem flexCol", o),
      "aria-readonly": x,
      "aria-disabled": p,
      variant: "body",
      weight: "default",
      component: "label",
      children: [
        /* @__PURE__ */ b("div", { className: d("flexRow alignCenter"), children: [
          /* @__PURE__ */ n(
            $,
            {
              "aria-labelledby": `${e}-label`,
              "aria-describedby": a ? `${e}-description` : null,
              value: s,
              isReadOnly: x,
              isDisabled: p,
              name: L,
              onChange: (y, k, g) => {
                typeof c?.onChange == "function" && c.onChange(y, k, g), typeof t == "function" && t(y, k, g);
              },
              ...m
            }
          ),
          /* @__PURE__ */ n(
            h,
            {
              id: `${e}-label`,
              variant: "body",
              component: "span",
              className: d("moonstone-checkboxItem_label"),
              children: l
            }
          )
        ] }),
        a && /* @__PURE__ */ n(
          h,
          {
            id: `${e}-description`,
            variant: "caption",
            weight: "default",
            component: "span",
            className: d("moonstone-checkboxItem_description flexRow"),
            children: a
          }
        )
      ]
    }
  );
};
C.displayName = "ControlledCheckboxItem";
const v = ({ defaultChecked: o = !1, onChange: e, ...s }) => {
  const [l, a] = _(o);
  return /* @__PURE__ */ n(
    C,
    {
      ...s,
      checked: l,
      onChange: (r, i) => {
        const t = !l;
        a(t), typeof e == "function" && e(r, i, t);
      }
    }
  );
};
v.displayName = "UncontrolledCheckboxItem";
const O = ({ checked: o, ...e }) => typeof o > "u" ? /* @__PURE__ */ n(v, { ...e }) : /* @__PURE__ */ n(C, { checked: o, onChange: e.onChange, ...e });
O.displayName = "CheckboxItem";
export {
  O as C,
  h as T,
  D as a,
  $ as b,
  N as c
};
